pkgs/servers/news/sn/default.nix

```
@@ -0,0 +1,33 @@
+{ lib, stdenv, fetchurl, zlib, ucspi-tcp }:
+
+let
+  pname = "sn";
+  version = "0.3.8";
+in stdenv.mkDerivation {
+
+  inherit pname version;
+
+  src = fetchurl {
+    url =
+      "https://www.ibiblio.org/pub/Linux/system/news/transport/sn-0.3.8.tar.bz2";
+    sha256 = "sha256-jOfHfVh4FJk8on1bvyu4sq0TERmVthE9yRdIRZbb3fA=";
+  };
+
+  configureFlags = [ ];
+
+  prePatch = ''
+    mkdir -p $out/sbin
+    mkdir -p $out/man/man8
+    sed -i s,/usr/local,$out, Makefile
+  '';
+
+  buildInputs = [ zlib ucspi-tcp ];
+
+  meta = {
+    homepage = "https://patrik.iki.fi/sn/";
+    description = "A small news system for small sites";
+    license = lib.licenses.gpl2Only;
+    platforms = lib.platforms.unix;
+    maintainers = [ lib.maintainers.ne9z ];
+  };
+}
```

```suggestion
{ lib
, stdenv
, fetchurl
, ucspi-tcp 
, zlib
}:

stdenv.mkDerivation (finalAttrs: {
  pname = "sn";
  version = "0.3.8";
  
  src = fetchurl {
    url = "https://www.ibiblio.org/pub/Linux/system/news/transport/sn-${finalAttrs.version}.tar.bz2";
    hash = "sha256-jOfHfVh4FJk8on1bvyu4sq0TERmVthE9yRdIRZbb3fA=";
  };

  postPatch = ''
    substituteInPlace Makefile \
      --replace  '/usr/local' '${placeholder "out"}'
  '';

  buildInputs = [ 
    ucspi-tcp
    zlib
  ];
  
  preInstall = ''
    mkdir -p $out/sbin $out/man/man8
  '';

  meta = {
    homepage = "https://patrik.iki.fi/sn/";
    description = "A small news system for small sites";
    license = lib.licenses.gpl2Only;
    maintainers = [ lib.maintainers.ne9z ];
    platforms = lib.platforms.unix;
  };
})
```

And put this in another PR.